### ISSUE TITLE
AF stopped animation [REopened]

### ISSUE THREAD

**Issue by Bodyash** [InventivetalentDev/AnimatedFrames#14]
## What version of the plugin are you using? *Type /version &lt;Plugin Name&gt;*  
`latest versions of AF, MM and PLAPI`
## What Spigot version are you using? *Type /version*  
`1.10.2`
## What plugins are you using? *Type /plugins*  
![default](https://cloud.githubusercontent.com/assets/2454569/16891257/b5f7e314-4aff-11e6-9cf0-0d5eea0f9421.PNG)
## Do you have an error log? Use [pastebin.com](http://pastebin.com). *If you're not sure, upload your whole server log*  
`No Errors`
## Additional information?  *(Are you using Bungeecord? Did it work in previous versions? etc.)*
No bungee. 

I didnt play all day. At night i log in and see what is happened. 2 of 7 animations stopped working.I was Told my friend about it. He installed Minecraft, created offline account, logged in and see the same as me. 2 of 7 frames not working. And ALL players on server says about it.

NEXT step what i do. Trying many ways to destroy working frames. Success, destroyed them, but when i put EMPTY item_frame to than wall - i see, that animation started working again. Wow.
![2016-07-16_02 30 56](https://cloud.githubusercontent.com/assets/2454569/16891321/b152cb3e-4b00-11e6-89c4-3c2f21618ce6.png)

I destroyed broken item_frames. put them back - nothing. 

I can put items in IT!
![2016-07-16_02 40 22](https://cloud.githubusercontent.com/assets/2454569/16891324/bb5506c4-4b00-11e6-827c-3b007d88f63a.png)

And i CANT PUT MAPS IN IT O_o

![2016-07-16_02 40 32](https://cloud.githubusercontent.com/assets/2454569/16891328/bfc6a7a8-4b00-11e6-8e5a-81c33eccef5c.png)

Only way to fix - restart server.

**Comment by InventivetalentDev** [comment]
Please upload your server log of that day/night.

**Comment by Bodyash** [comment]
There is fresh log: https://www.dropbox.com/s/pfa5kixub5e8djf/latest.log?dl=0

Sorry 4 errors in console, w8ting 4 LibsDisguises update. Stopped 2 animations:

![2016-07-17_00 28 19](https://cloud.githubusercontent.com/assets/2454569/16897298/986e9598-4bb5-11e6-91d3-b02fa363e114.png)
![2016-07-17_00 28 24](https://cloud.githubusercontent.com/assets/2454569/16897297/986e405c-4bb5-11e6-90ae-c7bf61f04554.png)

**Comment by InventivetalentDev** [comment]
There are multiple exceptions in the console, which are probably the reasons that the frames stop.
```
[16:35:35] [pool-15-thread-8/WARN]: Exception in thread "pool-15-thread-8" 
[16:35:35] [pool-15-thread-8/WARN]: java.util.ConcurrentModificationException
[16:35:35] [pool-15-thread-8/WARN]: 	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1429)
[16:35:35] [pool-15-thread-8/WARN]: 	at java.util.HashMap$KeyIterator.next(HashMap.java:1453)
[16:35:35] [pool-15-thread-8/WARN]: 	at org.inventivetalent.animatedframes.AnimatedFrame.run(AnimatedFrame.java:208)
[16:35:35] [pool-15-thread-8/WARN]: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[16:35:35] [pool-15-thread-8/WARN]: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[16:35:35] [pool-15-thread-8/WARN]: 	at java.lang.Thread.run(Thread.java:745)
```

```
[19:57:09] [pool-15-thread-3/WARN]: Exception in thread "pool-15-thread-3" 
[19:57:09] [pool-15-thread-3/WARN]: java.util.ConcurrentModificationException
[19:57:09] [pool-15-thread-3/WARN]: 	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1429)
[19:57:09] [pool-15-thread-3/WARN]: 	at java.util.HashMap$KeyIterator.next(HashMap.java:1453)
[19:57:09] [pool-15-thread-3/WARN]: 	at org.inventivetalent.animatedframes.AnimatedFrame.run(AnimatedFrame.java:208)
[19:57:09] [pool-15-thread-3/WARN]: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[19:57:09] [pool-15-thread-3/WARN]: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[19:57:09] [pool-15-thread-3/WARN]: 	at java.lang.Thread.run(Thread.java:745)
```

Just taking a wild guess: Those errors were probably also the cause for your last issue.  
I'll check the code for it tomorrow.

**Comment by Bodyash** [comment]
Wow, nice. Ty much!

**Comment by InventivetalentDev** [comment]
Should be fixed in v4.3.2.
